### PR TITLE
register fluid kerenls to phi [part 3]

### PR DESCRIPTION
**Pull request by huangjiyi**
### PR types

Others

### PR changes

Others

### Describe

Pcard-67000

Register fluid kernels of 29 ops to phi:
- cast
- c_broadcast
- c_concat
- c_embedding
- center_loss
- chunk_eval
- c_identity
- cinn_instruction_run
- cinn_launch
- clear_float_status
- coalesce_tensor
- collect_fpn_proposals
- conv_shift
- correlation
- cos_sim
- c_reduce_max
- c_reduce_min
- c_reduce_prod
- c_reducescatter
- c_reduce_sum
- crf_decoding
- cross_entropy
- c_scatter
- c_softmax_with_cross_entropy
- c_split
- c_sync_calc_stream
- c_sync_comm_stream
- ctc_align
- cudnn_lstm
- cvm

### CODE REVIEW

File: paddle/fluid/operators/collective/c_broadcast_op.cu.cc [hunk 1]
```diff
@@ -92,12 +92,14 @@ class CBroadcastOpCUDAKernel : public framework::OpKernel<T> {
...
+                          float,
+                          double,
 #if NCCL_VERSION_CODE >= 21000
-                        ops::CBroadcastOpCUDAKernel<plat::bfloat16>,
+                          plat::bfloat16,
 #endif
-                        ops::CBroadcastOpCUDAKernel<int>,
-                        ops::CBroadcastOpCUDAKernel<int64_t>,
-                        ops::CBroadcastOpCUDAKernel<plat::float16>);
+                          plat::float16) {
```
Comment (YuanRisheng) [inline paddle/fluid/operators/collective/c_broadcast_op.cu.cc:106]: 这里是不是缺了int和int64_t

File: paddle/fluid/operators/detection/collect_fpn_proposals_op.cc [hunk 2]
```diff
@@ -140,9 +140,11 @@ REGISTER_OPERATOR(
...
     paddle::framework::EmptyGradOpMaker<paddle::framework::OpDesc>,
     paddle::framework::EmptyGradOpMaker<paddle::imperative::OpBase>);
-REGISTER_OP_CPU_KERNEL(collect_fpn_proposals,
-                       ops::CollectFpnProposalsOpKernel<float>,
-                       ops::CollectFpnProposalsOpKernel<double>);
+PD_REGISTER_STRUCT_KERNEL(collect_fpn_proposals,
+                          CPU,
+                          ALL_LAYOUT,
+                          ops::CollectFpnProposalsOpKernel,
+                          float) {}
```
Comment (YuanRisheng) [inline paddle/fluid/operators/detection/collect_fpn_proposals_op.cc]: 这里是不是缺了double

File: paddle/fluid/operators/collective/c_broadcast_op.cu.cc (hunk 1 above)
Comment (huangjiyi) [reply paddle/fluid/operators/collective/c_broadcast_op.cu.cc:106]: Done, thanks ~

File: paddle/fluid/operators/detection/collect_fpn_proposals_op.cc (hunk 2 above)
Comment (huangjiyi) [reply paddle/fluid/operators/detection/collect_fpn_proposals_op.cc]: Done, thanks ~

**Review by YuanRisheng — DISMISSED** [review]

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by YuanRisheng — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**